### ISSUE TITLE
About experiments on CharadesSTA dataset

### ISSUE THREAD

**Issue by xljh0520** [jayleicn/moment_detr#11]
Hi, I noticed that you also conduct experiments on CharadesSTA dataset. I'm wondering how you prepare the video feature in CharadesSTA dataset? Could you share the feature files you prepared?

**Comment by jayleicn** [comment]
Hi @xljh0520, we don't have plans to release it, please check issue #8 for preparing CharadesSTA features, it should be quite straightforward. Let me know if you met any questions in the process.

**Comment by xljh0520** [comment]
> 

Hi, I have some questions in reproducing your experiments on CharadesSTA dataset. 

1. Did you change the CharadesSTA annotations into your QVH format?
2. If so, how to deal with the duration and relevant_windows? It's float in CharadesSTA but integer in your QVH.  
3. Did you use saliency loss in training CharadesSTA? If so how did you get the relevant_clip_ids in annotations ?

Thanks in advance!

**Comment by jayleicn** [comment]
Please check the attached [charades.zip](https://github.com/jayleicn/moment_detr/files/8305177/charades.zip) file, you should find your answers there.

**Comment by xljh0520** [comment]
> Please check the attached [charades.zip](https://github.com/jayleicn/moment_detr/files/8305177/charades.zip) file, you should find your answers there.

Thanks for your sharing! However, I train the model with the same hyperparameters in QVHdataset and I found the model always overfit and the R1@0.5 (0.3820) is far from the value you report (0.5363 ) in table 6.
Could you tell me the hyperparameters you set in CharadesSTA dataset？ Thank you！

**Comment by xljh0520** [comment]
> Hi @xljh0520, [opt.json.zip](https://github.com/jayleicn/moment_detr/files/8333458/opt.json.zip) contains a full list of hyperparameters saved by the training script. You may look into it to see which ones you need to change. There are a few important hyperparameters I am aware of: `lr`, `lw_saliency`, `lr_drop`, `clip_length`, and `n_epoch`.

Thanks for your kindly reply! I noticed another question. In [start_end_dataset.py#L97](https://github.com/jayleicn/moment_detr/blob/main/moment_detr/start_end_dataset.py#L97), it loads "saliency_pos_labels“ and "saliency_neg_labels" from meta["relevant_clip_ids"], meta["saliency_scores"]. However, there is no relevant_clip_ids and saliency_scores in annotations. 
So, I wanna ask about how you designed the saliency loss for the CharadesSTA dataset and the positive-negative video clip? Since I noticed that the 'lw_saliency' you set is 4.0, I suppose the saliency loss is important.

**Comment by jayleicn** [comment]
You can look into [start_end_dataset.py#L100](https://github.com/jayleicn/moment_detr/blob/main/moment_detr/start_end_dataset.py#L100) on how we perform pre-training using only ASR data. The ASR data contains only text queries and their start and end windows. Basically, we only use the `out` and `in` terms of the saliency loss described in `Equation 4` of the paper.

**Comment by xljh0520** [comment]
Thank you!

**Comment by Lonicer** [comment]
> Thank you!

Hello, how did you finally set up the Ground based on the Charades dataset?
What about relevant_clip_ids and saliency_scores in Truth? Thanks!

**Comment by YunjinYeung** [comment]
Hello, could you provide config files of training on TACoS dataset?
